src/main/java/Main.java

```diff
@@ -37,135 +44,188 @@
-            if ((rekontra != 't') && (rekontra != 'n')) throw new RuntimeException("Podano niepoprawna litere");
-        } else if (kontra != 'n') throw new RuntimeException("Podano niepoprawna litere");
+            reveto = scanner.next().charAt(0);
+            if ((reveto != 't') && (reveto != 'n'))
+                throw new RuntimeException("Podano niepoprawna litere");
+        } else if (veto != 'n')
+            throw new RuntimeException("Podano niepoprawna litere");
 
 
     }
 
-    private int obliczIloscZdobytychPunktow() {
-        int kolumnaStartowa;
-        int rob = 0;
-        int wiersz = iloscWylicytowanychLew - 1;
-        int suma = 0;
-        int tab[][] = tab3;
+    private int calculateMatchScore() {
+        int startingCol;
+        int offset = 0;
+        int row = amountOfBidedTricks - 1;
+        int sum = 0;
+        int supplementaryTable[][] = tableOfGameValuesBeforeMatch;
+
+        if (stageOfTheMatch.toLowerCase().equals("po"))
+            supplementaryTable = tableOfGameValuesAfterMatch;
+
+        if (suit == Suit.TREFL || suit == Suit.KARO)
+            startingCol = 4;
+        else if (suit == Suit.KIER || suit == Suit.PIK)
+            startingCol = 7;
+        else startingCol = 10;
+
+        if (veto == 't') {
+            if (reveto == 't')
+                offset = 2;
+            else offset = 1;
+        }
 
-        if (etap.toLowerCase().equals("po")) tab = tab4;
+        sum += supplementaryTable[row][startingCol + offset];
+        if (amountOfBloopers > 0)
+            sum -= amountOfBloopers * supplementaryTable[row][offset + 1];
+        else if (amountOfOvertricks > 0)
+            sum += amountOfOvertricks * supplementaryTable[MAX_AMOUNT_OF_TRICKS][startingCol + offset];
 
-        if (kolor == Kolor.TREFL || kolor == Kolor.KARO) kolumnaStartowa = 4;
-        else if (kolor == Kolor.KIER || kolor == Kolor.PIK) kolumnaStartowa = 7;
-        else kolumnaStartowa = 10;
 
-        if (kontra == 't') {
-            if (rekontra == 't') rob = 2;
-            else rob = 1;
-        }
-        suma += tab[wiersz][kolumnaStartowa + rob];
-        if (iloscWpadek > 0) suma -= iloscWpadek * tab[wiersz][rob + 1];
-        else {
-            if (iloscNadrobek > 0) suma += iloscNadrobek * tab[MAX_ILOSC_LEW][kolumnaStartowa + rob];
+        return sum;
+    }
 
+    private int getResultInInternationalMatchPoints(int matchScore, int valueOfHand) {
+        margin = matchScore - valueOfHand;
+        List<RangeOfIMP> listOfRanges = new ArrayList<>();
+        int scoreIMP = 0;
+
+        for (int i = 0; i <= LAST_ROW_IN_IMP_TABLE; i++) {
+            RangeOfIMP range = new RangeOfIMP(tableOfIMP[i][COL_WITH_IMP_RESULT - 2], tableOfIMP[i][COL_WITH_IMP_RESULT - 1], tableOfIMP[i][COL_WITH_IMP_RESULT]);
+            listOfRanges.add(range);
         }
 
-        return suma;
+        if (margin < (listOfRanges.get(0).highestValue - listOfRanges.get(0).lowestValue))
```

Comment:
Jako że to są zakresy wystarczy sprawdzić czy jest mniejsze od najmniejszego zakresu. (Akurat w tym wypadku by działało identycznie bo 40-20 = 20 ale przy innywhc wartoscaich program mógłby wariować)